GomokuClient/packages/gomoku-core/src/hooks/useJoinGame.ts

```diff
@@ -5,30 +5,34 @@ import { notification } from "@/shared/ui/notification";
 import { useRouter } from "@tanstack/react-router";
 import {
   SignalClientMessages,
-  SignalServerMessages,
   SignalDto,
+  SignalServerMessages,
   SwaggerTypes,
 } from "@/api";
 import { formatErrorMessage } from "@/utils/errorUtils";
 import { typedSessionStorage } from "@/shared/lib/utils";
 
 export const useJoinGame = (
   gameID: SwaggerTypes.CreateGameResponse["gameId"],
-  boardSize: SwaggerTypes.CreateGameResponse["boardSize"],
+  gameHistory: SwaggerTypes.GetGameHistoryResponse,
   playerID?: string,
 ) => {
   //TODO: refactor to separate hook or place inside tile hook
   const [moves, setMoves] = useState<string[]>([]);
   const [winningSequence, setWinningSequence] = useState<
-    SignalServerMessages.GameIsOverMessage["winningSequence"]
-  >([]);
+    SignalServerMessages.GameIsOverMessage["winningSequence"] | null
```

Comment:
I don't think we need extra null here as in SwaggerTypes it already genereted correctly

<img width="465" alt="Screenshot 2024-11-06 at 20 12 53" src="https://github.com/user-attachments/assets/4ddad881-700a-489b-a62d-bca8d60956c5">

Reply:
![image](https://github.com/user-attachments/assets/76c04c16-3bbe-445e-87d4-81d9d80f6e2d)

Reply:
This looks like bad trip. Would be nice to consolidate generated types

Reply:
I think | null is not required in generated type? Why we need to have both undefined and null in types?

Reply:
Okay, I will replace with GetGameHIstoryResponse usage for now and later we can figure out what to do